pkg/assembler/backends/inmem/hasSBOM.go

```diff
@@ -82,11 +98,57 @@ func (c *demoClient) IngestHasSBOMs(ctx context.Context, subjects model.PackageO
 	return modelHasSboms, nil
 }
 
-func (c *demoClient) IngestHasSbom(ctx context.Context, subject model.PackageOrArtifactInput, input model.HasSBOMInputSpec) (*model.HasSbom, error) {
-	return c.ingestHasSbom(ctx, subject, input, true)
+func (c *demoClient) IngestHasSbom(ctx context.Context, subject model.PackageOrArtifactInput, input model.HasSBOMInputSpec, includes model.HasSBOMIncludesInputSpec) (*model.HasSbom, error) {
+	funcName := "IngestHasSbom"
+
+	var softwareIDs []uint32
+	var dependencyIDs []uint32
+	var occurrenceIDs []uint32
+	var err error
+
+	if includes.Software != nil {
+		if softwareIDs, err = parseIDs(includes.Software); err != nil {
+			return nil, gqlerror.Errorf("%v ::  %s", funcName, err)
+		}
+		for _, id := range softwareIDs {
+			node, ok := c.index[id]
+			if ok {
+				if _, ok = node.(*pkgVersionNode); !ok {
```

Comment:
is there an alternative to this where it returns just the bool, the ok logic is a bit hard to read

Reply:
I'll extract this logic into a function

Reply:
done